src/Components/Components/src/RouteAttribute.cs

```diff
@@ -24,4 +24,10 @@ public RouteAttribute(string template)
     /// Gets the route template.
     /// </summary>
     public string Template { get; }
+    
+    /// <summary>
+    /// Gets or sets a flag to indicate whether the page should be rendered statically.
+    /// The effect of this flag is to suppress any <code>@rendermode</code> directives in the root component.
+    /// </summary>
+    public bool Static { get; set; }
```

Comment:
Declaring `Static` as a property of the route (as opposed to a property of the page) is more logically correct since:

1. It avoids any possibility of declaring "static" on a non-page component. The only place that static-ness can be stated is on the thing that makes it a page, i.e., the route.
2. In theory you can even have a page with multiple routes, some of which are static and some are interactive. That's some extra flexibility we now just get for free, and is logically sensible.

This does also necessitate also implementing the new Razor syntax though, since we don't want to force people to use `@attribute [Route("/url", Static = true)]`. I think the payoff of having this logical clarity is worth the cost of us having to do the syntax.

Reply:
Update: we discussed and concluded it's valid to do this in various ways, but the approach of declaring it on the route covers our needs, so I'll mark this as resolved.

Reply:
I'm a fan of including this property in the `RouteAttribute` in order to prevent any possible misusage where people go out of their way to try to add it to a non-routable component, but what was the resolution for custom routers? Is there a design that would make this work without necessitating changes to custom routers? Would making a lower-level API like `RouteTable` public help custom routers pick up this kind of changes automatically?

Reply:
I think custom routers are fine.

 * As long as your custom router sets the "static" flag on endpoint metadata, it will have the effect. People implementing custom routers are free to do that using `RouteAttribute` or any other mechanism that makes sense in their router.
 * As for how the interactive `Router` knows to exclude routes with this flag, a custom router can do the same thing if it wants, either by using `RouteAttribute` or whatever other mechanism makes sense for them.

Of course, if the router doesn't pay any attention this feature at all, then it won't apply with that router. But it couldn't "just work" with all possible routers that don't implement any logic for this, since they innately have to know not to route interactively to static components.